utils/mixins.py

```diff
@@ -46,29 +46,29 @@ def get_context_data(self, **kwargs: Any) -> dict:
 
         context_data = super().get_context_data(**kwargs)
 
-        from_date = None
-        to_date = None
-
-        if "year" in self.kwargs and "month" in self.kwargs:
-            year = int(self.kwargs["year"])
-            month = int(self.kwargs["month"])
-            current_date = timezone.now().date()
-            from_date = timezone.datetime(year=year, month=month, day=1).date()
-            if current_date.year == year and current_date.month == month:
-                to_date = current_date
-            else:
-                to_date = self._set_last_day_of_month_on_date(year=year, month=month)
-
-        if self.model == CustomUser:
-            user = self.object
-        else:
-            user = self.request.user
+        report_set = self.object.report_set.all() if self.model == CustomUser else self.get_queryset()
 
-        context_data["projects_work_percentage"] = user.get_projects_work_hours_and_percentage(from_date, to_date)
+        context_data["projects_work_percentage"] = self._get_projects_work_hours_and_percentage(report_set)
         return context_data
 
+    def _get_projects_work_hours_and_percentage(self, report_set: QuerySet) -> dict:
+        """
+        Returns dict where keys are Project names and values are lists containing total sum of work hours
+        and percentage amount of work by this user.
+        """
+        work_hours_per_project = report_set.values("project__name").annotate(total_hours=Sum("work_hours"))
+        work_hours_sum = work_hours_per_project.aggregate(Sum("total_hours"))["total_hours__sum"]
+        return self._get_dict_of_total_work_hours_per_project_statistics(work_hours_per_project, work_hours_sum)
+
     @staticmethod
-    def _set_last_day_of_month_on_date(year: int, month: int) -> date:
-        if month == 12:
-            return date(year=year, month=month, day=31)
-        return date(year=year, month=month + 1, day=1) - timezone.timedelta(days=1)
+    def _get_dict_of_total_work_hours_per_project_statistics(
+        queryset: QuerySet, work_hours_sum: timezone.timedelta
+    ) -> dict:
+        return {
+            annotation["project__name"]: (
+                [annotation["total_hours"], (annotation["total_hours"] / work_hours_sum) * 100]
+                if work_hours_sum.total_seconds() > 0
+                else 0
```

Comment:
Please extract this to separate private method and name it well

Reply:
Which section exactly? It's a short code and a fairly simple logic of what is already a private method, which name already explains what it does: `get dict of total work hours per project statistics`. If name isn't descriptive enough I could change it, but it's already a mouthful. I can also add a comment explaining what happens here.

Reply:
Done.